node/src/hopper/routing_service.rs

```diff
@@ -1869,6 +1860,113 @@ mod tests {
     }
 
+    fn route_data_to_peripheral_component_uses_proper_key_on_payload_for_component<F>(
+        payload_factory: F,
+        target_component: Component,
+    ) where
+        F: FnOnce(&CryptDEPair) -> CryptData,
+    {
+        let peer_actors = peer_actors_builder().build();
+        let subject = RoutingService::new(
+            make_cryptde_pair(),
+            RoutingServiceSubs {
+                proxy_client_subs_opt: peer_actors.proxy_client_opt,
+                proxy_server_subs: peer_actors.proxy_server,
+                neighborhood_subs: peer_actors.neighborhood,
+                hopper_subs: peer_actors.hopper,
+                to_dispatcher: peer_actors.dispatcher.from_dispatcher_client,
+                to_accountant_routing: peer_actors.accountant.report_routing_service_provided,
+            },
+            100,
+            200,
+            true,
+        );
+        let route = Route::single_hop(&PublicKey::new(b"1234"), subject.cryptdes.main).unwrap();
+        let payload = payload_factory(&subject.cryptdes);
+        let live_package = LiveCoresPackage::new(route, payload);
+
+        subject.route_data_to_peripheral_component(
+            target_component,
+            SocketAddr::from_str("1.2.3.4:1234").unwrap(),
+            live_package,
+            true,
+        );
+
+        // CryptDENull panics when you try decrypting with the wrong key; no panic means test passes
+    }
+
+    #[test]
+    fn route_data_to_peripheral_component_uses_alias_key_on_payload_for_proxy_client() {
+        let payload_factory = |cryptdes: &CryptDEPair| {
+            encodex(
+                cryptdes.alias,
+                cryptdes.main.public_key(),
+                &MessageType::ClientRequest(VersionedData::new(
+                    &crate::sub_lib::migrations::client_request_payload::MIGRATIONS,
+                    &ClientRequestPayload_0v1 {
+                        stream_key: StreamKey::new(
+                            PublicKey::new(b"1234"),
+                            SocketAddr::from_str("1.2.3.4:1234").unwrap(),
+                        ),
+                        sequenced_packet: SequencedPacket::new(vec![1, 2, 3, 4], 1234, false),
+                        target_hostname: Some("hostname".to_string()),
+                        target_port: 1234,
+                        protocol: ProxyProtocol::TLS,
+                        originator_alias_public_key: PublicKey::new(b"1234"),
+                    },
+                )),
+            )
+            .unwrap()
+        };
+        route_data_to_peripheral_component_uses_proper_key_on_payload_for_component(
+            payload_factory,
+            Component::ProxyClient,
+        );
+    }
+
+    #[test]
+    fn route_data_to_peripheral_component_uses_alias_key_on_payload_for_proxy_server() {
+        let payload_factory = |cryptdes: &CryptDEPair| {
+            encodex(
+                cryptdes.alias,
+                cryptdes.alias.public_key(),
```

Comment:
Also I really hate these two distinguished arguments as by logic they should allow a merge of them into one. 
I know there might be a catch somewhere out here but I would like to at least try if it is reasonable or just a silly idea that we keep alive.